packages/esm-form-entry-app/src/app/services/concept.service.ts

```diff
@@ -19,50 +22,39 @@ export class ConceptService {
 
   constructor(private http: HttpClient, private windowRef: WindowRef) {}
 
-  public getUrl(): string {
-    return this.windowRef.openmrsRestBase + 'concept';
-  }
-
-  public searchConceptByUUID(conceptUUID: string, lang: string): Observable<any> {
-    return this.http.get(this.getUrl() + `/${conceptUUID}?v=full&lang=${lang}`, {
-      headers: this.headers,
-    });
-  }
-
-  public searchBulkConceptsByUUID(conceptUuids: Array<string>, lang: string): Observable<Array<ConceptMetadata>> {
-    const observablesArray = [];
-    const relativeConceptLabelUrls = ConceptService.getRelativeConceptLabelUrls(conceptUuids, lang);
-
-    for (const relativeConceptLabelUrl of relativeConceptLabelUrls) {
-      observablesArray.push(
+  public searchConceptsByIdentifiers(conceptIdentifiers: Array<string>) {
+    return of(...ConceptService.getConceptReferenceUrls(conceptIdentifiers)).pipe(
+      map((referenceUrl) =>
         this.http
-          .get<ListResult<ConceptMetadata>>(this.windowRef.openmrsRestBase + relativeConceptLabelUrl, {
+          .get<ConceptReferencesResult>(this.windowRef.openmrsRestBase + referenceUrl + '?v=custom:(uuid,display)', {
```

Comment:
I think that the `?v=custom...` part should also be returned by `getConceptReferenceUrls`. The reason for this is the offline precaching: Without having the query param in the "URL set", the setup function will register "wrong" URLs and won't precache them.

Reply:
That's a good point... Also, the `?` is wrong here anyway, though in a way that's only harmful to the amount of data sent over the wire.